pkg/build/builder/docker.go

```diff
@@ -65,7 +65,7 @@ func (d *DockerBuilder) Build() error {
 		return err
 	}
 	defer removeImage(d.dockerClient, imageTag(d.build))
-	if d.build.Parameters.Output.Registry != "" || d.authPresent {
+	if len(d.build.Parameters.Output.DockerImageReference) != 0 {
```

Comment:
i'm guessing @mfojtik 's sti builder builder custom image will need to consume this change, and anyone else who's produced a custombuilder so far (RCM?)  So does this not fall into the API backwards compatibility requirements?